src/packages/galaxy/galaxy.py

```diff
@@ -80,15 +80,34 @@ def __init__(self, url: str = None):
         data = await self._call("api/systems", {"filter[name:eq]": name.upper()})
         result_count = data['meta']['results']['available']
         if result_count > 0:
+            system_id = data['data'][0]['id']
             sys = data['data'][0]['attributes']
-            main_star = await self._call("api/stars",
-                                         {"filter[systemId64:eq]": sys['id64'],
-                                          "filter[isMainStar:eq]": 1})
-            if main_star['meta']['results']['available'] > 0:
-                sys['spectral_class'] = main_star['data'][0]['attributes']['subType'][0]
+            main_star = await self._find_main_star(system_id)
+            sys['spectral_class'] = main_star['spectral_class'] if main_star is not None else None
             return StarSystem(position=Vector(**sys['coords']),
                               name=sys['name'],
-                              spectral_class=sys.get('spectral_class'))
+                              spectral_class=sys['spectral_class'])
+
+    async def _find_main_star(self, system_id: int) -> Optional[Dict]:
```

Comment:
[Nitpick] Does this *actually* return None if the system isn't found?  Where does that happen?

Reply:
The `return None` happens implicitly with Python if the function reaches the end without an explicit `return`.

This case is also tested with `test_find_system_by_invalid_name`, asserting the result is `None`.

Reply:
Good enough for me.  Thanks.